recipes/mountbroker.rb

```diff
@@ -0,0 +1,44 @@
+#
+# Cookbook Name:: gluster
+# Recipe:: mountbroker
+#
+# Copyright 2016, Yakara Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+include_recipe 'gluster::geo_replication_install'
+require 'json'
+
+mb = node['gluster']['mountbroker']
+
+group mb['group'] do
```

Comment:
The group existing or not is determined in the compile phase.  While a group being created will happen in the converge phase:

- Wrapper determines group does not exist and needs to be created
- Gluster determines group does not exist and needs to be created
- Wrapper creates group
- Gluster tries to create group and bombs out